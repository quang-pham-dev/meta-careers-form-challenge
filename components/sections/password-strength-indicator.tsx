import { CheckCircle2, XCircle } from 'lucide-react'

interface PasswordStrengthIndicatorProps {
  label: string
  met: boolean
}

export function PasswordStrengthIndicator({
  label,
  met,
}: PasswordStrengthIndicatorProps) {
  return (
    <div
      className={`flex items-center ${met ? 'text-green-500' : 'text-red-500'}`}>
      {met ? (
        <CheckCircle2 className="h-4 w-4 mr-2" aria-label="Check" />
      ) : (
        <XCircle className="h-4 w-4 mr-2" aria-label="X" />
      )}
      <span className="text-sm">{label}</span>
    </div>
  )
}
