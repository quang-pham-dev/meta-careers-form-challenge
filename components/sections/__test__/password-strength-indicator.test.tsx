import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { PasswordStrengthIndicator } from '../password-strength-indicator'

describe('PasswordStrengthIndicator', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <PasswordStrengthIndicator label="Test Label" met={false} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the label correctly', () => {
    render(<PasswordStrengthIndicator label="Test Label" met={false} />)
    expect(screen.getByText('Test Label')).toBeInTheDocument()
  })

  it('displays a check icon when condition is met', () => {
    render(<PasswordStrengthIndicator label="Test Label" met={true} />)
    expect(screen.getByLabelText('Check')).toBeInTheDocument()
  })

  it('displays an X icon when condition is not met', () => {
    render(<PasswordStrengthIndicator label="Test Label" met={false} />)
    expect(screen.getByLabelText('X')).toBeInTheDocument()
  })

  it('applies correct text color when condition is met', () => {
    render(<PasswordStrengthIndicator label="Test Label" met={true} />)
    expect(screen.getByText('Test Label').closest('div')).toHaveClass(
      'text-green-500',
    )
  })

  it('applies correct text color when condition is not met', () => {
    render(<PasswordStrengthIndicator label="Test Label" met={false} />)
    expect(screen.getByText('Test Label').closest('div')).toHaveClass(
      'text-red-500',
    )
  })
})
