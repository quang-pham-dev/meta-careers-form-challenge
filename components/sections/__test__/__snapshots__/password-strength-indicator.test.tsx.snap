// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordStrengthIndicator > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center text-red-500"
  >
    <svg
      aria-label="X"
      class="lucide lucide-circle-x h-4 w-4 mr-2"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="m15 9-6 6"
      />
      <path
        d="m9 9 6 6"
      />
    </svg>
    <span
      class="text-sm"
    >
      Test Label
    </span>
  </div>
</DocumentFragment>
`;
